app/src/main/java/com/example/firstapp/MQTT/MqttHandler.kt

```diff
@@ -109,19 +116,40 @@ class MqttHandler : AppCompatActivity {
                 @Throws(Exception::class)
                 override fun messageArrived(topic: String, message: MqttMessage) {
                     if (topic == CAMERA_SUB) {
-                        setCameraView(message)
+                        val bm =
+                            Bitmap.createBitmap(IMAGE_WIDTH, IMAGE_HEIGHT, Bitmap.Config.ARGB_8888)
+                        val payload = message.payload
+                        val colors = IntArray(IMAGE_WIDTH * IMAGE_HEIGHT)
+                        for (ci in colors.indices) {
+                            val r = payload[3 * ci]
+                            val g = payload[3 * ci + 1]
+                            val b = payload[3 * ci + 2]
+                            colors[ci] = Color.rgb(r.toInt(), g.toInt(), b.toInt())
+                        }
+                        bm.setPixels(colors, 0, IMAGE_WIDTH, 0, 0, IMAGE_WIDTH, IMAGE_HEIGHT)
+                        mCameraView?.setImageBitmap(bm)
                     }
                     if (topic == TRAVELED_DIS) {
-                        setDistanceView(message)
+                        val distance = message.toString()
+                        mTraveledDistance?.setText(distance + " m")
                     }
                     if (topic == ULTRASOUND_SUB) {
-                        setWarningView(message)
+                        val ultraSound = message.toString()
+                        if (ultraSound > 1.toString()) {
+                            mFront?.setText("WARNING")
+                            mFront?.setTextColor(RED)
+                        } else {
+                            mFront?.setText("")
+                        }
                     }
                     if (topic == SPEED_SUB) {
-                        setSpeedView(message)
+                        val speed = message.toString()
+                        mSpeed?.setText(speed)
                     }
                     if (topic == BIN_CAPACITY) {
-                        setBinView(message)
+                        val bagful = message.toString()
+                        mBagCapacity?.setText("${bagful}% ")
```

Comment:
I think old code might have replaced the new one in master during a merge.  The ones marked in red, are the refatored code in master, while the green is the old one.

Reply:
Ok . I will check again and push it